crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -128,6 +128,16 @@ impl<'w> EntityRef<'w> {
     }
 }
 
+impl<'w> From<EntityMut<'w>> for EntityRef<'w> {
+    fn from(entity_mut: EntityMut<'w>) -> EntityRef<'w> {
+        EntityRef {
+            world: entity_mut.world,
+            entity: entity_mut.entity,
+            location: entity_mut.location,
+        }
```

Comment:
I'd much rather that this use the safe path.

```
let entity = self.entity;
self.world().get_entity_or_whatever_the_method_is(entity)
```

In that case, I think we should have it as a method instead.

Reply:
Can you clarify? Are you asking to use EntityRef::new, or something else entirely?
It seems as though the alternative "safe path" is just doing another lookup for no reason. Why is it "safe" and the proposed change is not?

Reply:
We definitely should use `EntityRef::new` (which also should probably be marked `unsafe`, since the `Location` being 'valid' is "relied" upon for safety).

Looking more closely (not on mobile), I do think we can get away with avoiding the other lookup.

I'm still not sure it should be a `From` impl either really, given that it should still be valid so long as it takes an exclusive refernece to self.

Reply:
Please provide justifications for your proposals

Reply:
Yeah; I'm not sure I agree with you here. The invariants of `EntityMut` and `EntityRef` will always be tied closely together; converting into the immutable form should always be straightforward and safe.

Reply:
> I'm still not sure it should be a From impl either really, given that it should still be valid so long as it takes an exclusive refernece to self.

This would allow `&'e EntityMut<'w> -> EntityRef<'e>`, which requires the `EntityMut` to remain alive, while what the PR proposes would add `EntityMut<'w> -> EntityRef<'w>`, consuming the `EntityMut` thus not requiring it to be alive afterwards. They are different usecases IMO.

Reply:
@DJMcNab can this be marked as resolved?

Reply:
I still don't like that it uses the struct syntax, but I can see that others disagree. I do agree that this entire module has insufficient safety comments, but I hardly see how that's justification to make it worse.
But like I say, that seems to be fighting against the tide.

Fwiw, I'm still not convinced that the owned to owned path is the most useful/idiomatic, or that it should be a From impl. I know it's not quite the same, but `std::sync::Weak` doesn't have a `From<std::sync::Arc>` impl.

But again, I've been overruled.

I guess we can resolve it, although I can't say I'm super pleased.

Reply:
I agree on using `EntityRef::new` rather than building the struct manually. It would help to not miss when adding controls/safety to the `new` later

Reply:
I'm convinced; let's go with `EntityRef::new`.

Reply:
0163bdd